### ISSUE TITLE
The opencert website is not fully accessible from China

### ISSUE THREAD

**Issue by mohammedkalimullah** [OpenCerts/opencerts-website#542]
Hi, 
The opencert.io site cannot be fully accessible in China and some of the website components are timeout.

The students/employers cannot able verify the certificate from china. could you please help advise on this issue?

[opencertsissue_china.docx](https://github.com/OpenCerts/opencerts-website/files/5054378/opencertsissue_china.docx)

**Comment by Nebulis** [comment]
If China is blocking something there is nothing much we can do on our side. Are you able:
- to show us how it's working when accessing from China.
- to give us more details on what's blocking.

I can eventually give better guidelines if you are not sure how to share those information.

**Comment by mohammedkalimullah** [comment]
The website itself is not loading completely. it takes more time to load and some of the components are being timeout. like i have shared in the previous attachments.

**Comment by Nebulis** [comment]
Ah sorry I thought it was a certificate. From the doc it's hard to say what it taking long / what is blocked. Is it possible to share the content of the network panel ?

**Comment by mohammedkalimullah** [comment]
[opencertsissue_china.docx](https://github.com/OpenCerts/opencerts-website/files/5054620/opencertsissue_china.docx)
Please find the attached information.

**Comment by Nebulis** [comment]
List of failed resources:
- fontawsome
- jquery (why ???????)
- popper
- bootstrap

This could be fixed by
- using chinese cdn when serving from china (https://docs.netlify.com/routing/redirects/redirect-options/#redirect-by-country-or-language)
- get rid of dependencies or add them into the bundle (why do we have jquery ?)

Blockers:
- Are you able to tell if Infura (which we use to verify the status of a cert) is also blocked ?
- How to test it's working correctly ?

**Comment by Nebulis** [comment]
For infura, maybe you can still drag and drop / open a certificate, if that's the case just check the network panel again

**Comment by mohammedkalimullah** [comment]
do you mean the jquery is from renderer. From the network tab,it is from opencert website.

Is this issue is because of the renderer site? FYI. we are not using the netlify to host our renderer. we use our internal server to host due to security.

**Comment by Nebulis** [comment]
No in this case the note is for myself => why do we use jquery ? I dont think it's needed. Feel free to ignore that part

However your comments made me realize something else. The problem could happen with renderer as well (not only opencerts) and in this case we have no control over it (so if a renderer use a lib that is blocked by China then you will need to reach out the owner of the renderer)

That aside we can still figure out from opencerts perspective what is needed.

Did you figure out for infura ? I googled a bit and it looks ok in China (doesn't mean it will work forever), but I have no way to check by myself

**Comment by mohammedkalimullah** [comment]
Thanks for your reply. i noticed all the blocks is from opencert website not from the renderer site.

Not sure what we need to check from renderer site.

Regarding infura, I will check with student from china again and let you know.

**Comment by Nebulis** [comment]
From the renderer it will be the same problem, if  you use a cdn that is blocked in China to load assets (css,js)

**Comment by mohammedkalimullah** [comment]
I still dont understand. what i am referring is that before i reach renderer to view cert, we are facing those errors.
Please advise on the action item to solve this issue.

**Comment by Nebulis** [comment]
China fire wall can create 2 problems:
1. with opencerts where some assets are not loaded, eventually infura is blocked and so on.
1. with decentralised renderer where some assets could also not be loaded.

On our side we can work on the first problem (i.e. with opencerts), but it's also worth to note that it's not as simple as it sounds, so this problem wont be solved today or this week>

On your side, if you developed a renderer you can eventually start to check if your renderer is not blocked by China. If you didn't develop a renderer but uses one, then you can check directly with the developer of the renderer.

**Comment by mohammedkalimullah** [comment]
Thanks for reply.

For problem 1, Please advice on the approximate timeline, so that we can update our users.
For renderer , we use the Decentralised Document Renderer as per opencerts documentation.

**Comment by Nebulis** [comment]
- no timeline
- even if you follow the provided renderer as a base, maybe you added bootstrap? maybe you added something else ? If you didn't add anything I guess it's ok but don't take this for granted.

**Comment by gjj** [comment]
Great firewall of China's tester, for future reference:

- FontAwesome https://en.greatfire.org/https/use.fontawesome.com
- jQuery https://en.greatfire.org/https/code.jquery.com
- Infura https://en.greatfire.org/https/infura.io

**Comment by mohammedkalimullah** [comment]
can we have any update please? .we also received feedback from students that it takes 2-3 minutes if we access from US location.

**Comment by Nebulis** [comment]
We will discuss about that with business owner on Tuesday

**Comment by mohammedkalimullah** [comment]
Thanks! Let us know once you have any update

**Comment by Nebulis** [comment]
We just discussed about that, the goal is to provide support on it but we have other priorities to work on. so it might take time (goal is this be be fulfilled before end of year)

Also as we don't have way to test from China it would be nice if you could help us (for the testing part)

**Comment by Nebulis** [comment]
#569 will fix the following issues:
- jquery
- popper
- bootstrap

**Comment by mohammedkalimullah** [comment]
Thanks for the update.
can we ask the user from China to retry again?

**Comment by Nebulis** [comment]
Not yet it's not deployed, I will let you know

**Comment by Nebulis** [comment]
@mohammedkalimullah Could you ask the user to try again ?

**Comment by mohammedkalimullah** [comment]
Just got the feedback from user and still experiencing the same issue.

**Comment by Nebulis** [comment]
Ok thanks. Are your user able to make a video of opencerts verification with the chrome console open and share the video to us ?

**Comment by mohammedkalimullah** [comment]
I will try to request from user. 
meanwhile i checked using the below link and it seems the fontawsome is not loading.
https://www.site24x7.com/web-page-analyzer.html
![image](https://user-images.githubusercontent.com/53372610/98495375-f4e57e00-2279-11eb-9734-1ed5842a48f6.png)

**Comment by Nebulis** [comment]
Yeah it's listed above but it shouldn't block the website from working. It's just icons

**Comment by mohammedkalimullah** [comment]
![image](https://user-images.githubusercontent.com/53372610/98501019-02eecb00-2289-11eb-8acf-1a2724b89c57.png)
There are other components not loaded as well

**Comment by mohammedkalimullah** [comment]
![image](https://user-images.githubusercontent.com/53372610/98501058-20239980-2289-11eb-98c4-f417d68e7fe2.png)

**Comment by Nebulis** [comment]
Am I right to day that nothing is loaded at all? I see that everything size is 0b.. 

It's different than before in that case it used to load

**Comment by mohammedkalimullah** [comment]
Yes. it seems sometimes it doesnt load anything,
e.g. i tried to access the below URL location. it doesnt load
![image](https://user-images.githubusercontent.com/53372610/98509651-8960d800-229c-11eb-887f-471f1d1bfdb8.png)

**Comment by Nebulis** [comment]
This might get more troublesome then ... I see someone reported some instability as well https://community.netlify.com/t/site-blocked-in-china-netlify-and-the-chinese-great-firewall-gfw/4259/17

Could you provide me a HAR => https://support.zendesk.com/hc/en-us/articles/204410413-Generating-a-HAR-file-for-troubleshooting
